### ISSUE TITLE
Regression in file module

### ISSUE THREAD

**Issue by ghost** [ansible/ansible#7664]
##### Issue Type:

Bug Report
##### Ansible Version:

devel, commit 202ac68995ecd8d7284c7e17ff79ca1d8a946102
##### Environment:

Virtualenv with ansible-1.7 devel, managing Ubuntu 14.04
##### Summary:

Regression in file state=link
##### Steps To Reproduce:

Use the file module to create a symlink on a remote machine, using absolute paths.
##### Expected Results:

rstrox-apache-1          : ok=1    changed=1    unreachable=0    failed=0
##### Actual Results:

```
% cat demobugs.yml
#!/usr/bin/env ansible-playbook 

---
- name: Demonstrate bug in 202ac68995ecd8d7284c7e17ff79ca1d8a946102
  serial: 1
  hosts: apache
  gather_facts: true

  tasks:
    - name: symlink
      file: >
        state=link
        src=/etc/apache2/mods-available/{{ item }}.load
        dest=/etc/apache2/mods-enabled/{{ item }}.load
      with_items:
        - rewrite
        - proxy


 % ./demobugs.yml -vvv

PLAY [Demonstrate bug in 202ac68995ecd8d7284c7e17ff79ca1d8a946102] ************ 
...
...
...
TASK: [symlink] *************************************************************** 
<23.XXX.XXX.XXX> ESTABLISH CONNECTION FOR USER: root
<23.XXX.XXX.XXX> REMOTE_MODULE file state=link src=/etc/apache2/mods-available/rewrite.load dest=/etc/apache2/mods-enabled/rewrite.load
<23.XXX.XXX.XXX> EXEC ['ssh', '-C', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/rstrox/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'StrictHostKeyChecking=no', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=root', '-o', 'ConnectTimeout=10', u'23.XXX.XXX.XXX', u"/bin/sh -c 'LC_CTYPE=en_US.UTF-8 LANG=en_US.UTF-8 /usr/bin/python'"]
failed: [rstrox-apache-1] => (item=rewrite) => {"failed": true, "item": "rewrite", "path": "/etc/apache2/mods-enabled/rewrite.load", "src": "/etc/apache2/mods-available/rewrite.load", "state": "absent"}
msg: src file does not exist, use "force=yes" if you really want to create the link: /etc/apache2/mods-enabled/rewrite.load
...
...
...

% ssh rstrox-apache-1
Welcome to Ubuntu 14.04 LTS (GNU/Linux 3.13.0-24-generic x86_64)
...
root@rstrox-apache-1:~# ls -l /etc/apache2/mods-available/rewrite.load
-rw-r--r-- 1 root root 66 Jan  3 14:48 /etc/apache2/mods-available/rewrite.load
root@rstrox-apache-1:~# ls -l /etc/apache2/mods-enabled/rewrite.load
ls: cannot access /etc/apache2/mods-enabled/rewrite.load: No such file or directory

```

**Comment by jimi-c** [comment]
I have created the above patch to resolve this, if you'd like to test it. I've run it against the sample play you've included and it resolves the issue there.

**Comment by ghost** [comment]
Cool, looks good to me. Thanks for the quick resolution!

**Comment by jimi-c** [comment]
Great, I've gone ahead and merged this in. Please let us know if you continue seeing any problems regarding this issue. Thanks!

**Comment by rouge8** [comment]
@jimi-c I'm seeing this error in 1.6.10.

**Comment by jimi-c** [comment]
@rouge8 this will be included in 1.7.
